### ISSUE TITLE
Impossible to playtest Arena maps in GB

### ISSUE THREAD

**Issue by PrOF-kk** [Gang-Garrison-2/Gang-Garrison-2#226]
Fairly self-explanatory, here's how to replicate it quickly:
- Start GG2 and select Garison Builder, then Arena
- Load a map, the first you'll find is probably arena_harvest, but anyone will do
- Test w/o save > Test here

Enjoy not being able to test your map

**Comment by Medo42** [comment]
To be fair, everything is kind of working as specified ;)

Should arena maps generally let you run around freely if you're alone on the server, or should the behavior be changed specifically for the GB testing case?

**Comment by PrOF-kk** [comment]
I think it would be more elegant letting the player roam around in general, but it's fine either way

**Comment by Medo42** [comment]
I just spent several hours picking apart the Arena game mode logic, and I finally kind of have the picture where everything fits and managed to mostly get this working (allow spawning and running around in the "waiting for players" state), but it's not quite there yet (for example, you don't automatically respawn when the server enters "waiting for players" again, several variables like humiliation are not reset, some hint texts about respawning are wrong now, and I'm sure some other things are off as well).
The whole gamemode logic is written in a way that makes it a *lot* harder to understand and modify than is necessary. Since I took the time now to dig into it, I'll also go the next step and rewrite it into something that should be a lot easier to wrap ones head around. The idea is to make it more closely resemble a state machine (states here being "waitingForPlayers", "roundSetup", "roundProper", "roundEnd" and perhaps "mapEnd").
